walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,112 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.Enumeration;
+import java.util.Map;
+import java.util.Iterator;
+
+import java.util.Comparator;
 
 public final class FindMeetingQuery {
+  /** Return free slots based on list of time points. */
+  public Collection<TimeRange> getFreeSlots(List<TimePoint> importantEvents, MeetingRequest request) {
+    Collection<TimeRange> possibleSlots = new ArrayList<>();
+    // Sort all time points.
+    Collections.sort(importantEvents, TimePoint.ORDER_END_START);
+
+    // Number of active meeting for the moment under consideration.
+    int numberOfActiveMeetings = 0;
+    int endOfLastMeeting = 0;
+
+    // numberOfActiveMeetings will be 0  - before start and after end of the day,
+    // 1 - if current time slot is free for all mandatory attendees,
+    // >1 - in any other case.
+  
+    // numberOfActiveMeetings will increase if new time point under consideration is a start for a meeting, and decrease in other case.
```

Comment:
nit: some of the comments look very long. Please break them into multiple lines so that no lines are more than 100 characters. Perhaps this is something that can be automatically configured on the IDE you are using.

Reply:
I skimmed through all of my editor's preferences but didn't find anything like that, but from now on I'll do it manually.

Reply:
Not needed now, but for the capstone project you can try using https://github.com/google/google-java-format to automatically format all changed java files.

Reply:
Also, if you are using IntelliJ, there is this: https://github.com/google/styleguide/blob/gh-pages/intellij-java-google-style.xml